MaxMind.Db/Reader.cs

```diff
@@ -170,8 +212,75 @@ private void Dispose(bool disposing)
         /// <returns>An object containing the IP related data</returns>
         public T Find<T>(IPAddress ipAddress, InjectableValues injectables = null) where T : class
         {
-            int prefixLength;
-            return Find<T>(ipAddress, out prefixLength, injectables);
+            return Find<T>(ipAddress, out _, injectables);
+        }
+
+        /// <summary>
+        /// Get an enumerator that iterates all data nodes in the database
+        /// </summary>
+        /// <param name="cacheSize">The size of the data cache. This can greatly speed enumeration at the cost of memory usage.</param>
+        /// <returns>Enumerator for all data nodes</returns>
+        public IEnumerable<Reader.ReaderIteratorNode<T>> FindAll<T>(int cacheSize = 16384) where T : class
```

Comment:
A test that used something other than `Dictionary<string, object>` might be good. In particular, one using the `TypeHolder` class with the `MaxMind-DB-test-decoder.mmdb` database.